### PR TITLE
Better logging when files are accessed.

### PR DESCRIPTION
**Pull request by jhdulaney**
## Description
This change increases the information stored when a datafile key is created, such as the ip address causing the creation, project id, and access token.

This change also stores all this information, plus key creation timestamp and the key itself, in the datafile access log whenever a datafile is accessed.

## Related Issue
n/a

## Testing
* passed automated testing
* tested basic file download functionality
  - download of public files while logged out
  - download of public files while logged in
  - download of private files while logged in
  - download of private files while logged out but with key
* verified that each of the above tests produced the correct logs

Signed-off-by: Mairi Dulaney <jdulaney@fedoraproject.org>

### CODE REVIEW

File: data_import/models.py [hunk 1]
```diff
@@ -117,18 +124,23 @@ def file_url_as_attachment(self):
         """
         return self.file.storage.url(self.file.name)
 
-    @property
```
Comment (madprime) [inline data_import/models.py]: It's funny, I wonder if this was intended to be used in templates (the way `download_url` is! But I don't think it ever was. Removing this seems to be fine.

File: data_import/serializers.py [hunk 2]
```diff
@@ -8,10 +11,33 @@ class DataFileSerializer(serializers.ModelSerializer):
...
 
-    download_url = serializers.CharField(source='private_download_url')
     metadata = serializers.JSONField()
 
     class Meta:  # noqa: D101
         model = DataFile
         fields = ('id', 'basename', 'created', 'download_url', 'metadata',
                   'source')
+
+    def to_representation(self, instance):
```
Comment (madprime) [inline data_import/serializers.py]: Is it possible to...
1. override `_readable_fields` to `super()` then remove `download_url` from the returned set
2. super() `to_representation`, then add `download_url`
?

Thinking about how to do this in a way that minimizes customization on REST framework :)

File: data_import/views.py [hunk 3]
```diff
@@ -46,12 +71,9 @@ def get(self, request, *args, **kwargs):
...
             return self.get_and_log(request)
 
-        query_key = request.GET.get('key', None)
-        if query_key:
-            key_qs = DataFileKey.objects.filter(datafile=self.data_file)
-            key_qs = key_qs.filter(key=query_key)
-            if key_qs.exists():
-                if not key_qs.get().expired:
-                    return self.get_and_log(request)
+        key = self.get_key(request)
```
Comment (madprime) [inline data_import/views.py]: redundant?

File: data_import/models.py (hunk 1 above)
Comment (jhdulaney) [reply data_import/models.py]: Yeah, I grepped for it and only found it in two places, here in the model and in the serializer.  I'm certain it's safe to remove.

File: data_import/serializers.py (hunk 2 above)
Comment (jhdulaney) [reply data_import/serializers.py]: The docs do specifically state:
There are four methods that can be overridden, depending on what functionality you want the serializer class to support:

    .to_representation() - Override this to support serialization, for read op

So, doing it this way seems to be more in line with the way that rest-framework wants things done?

File: data_import/views.py (hunk 3 above)
Comment (jhdulaney) [reply data_import/views.py]: oh, lol

File: data_import/serializers.py (hunk 2 above)
Comment (madprime) [reply data_import/serializers.py]: But this code seems like it's a copy/paste of some version of `ModelSerializer`... but not the current `master`. And it's relying on `_to_representation` having a particular behavior, so... I don't think this makes less assumptions than my suggestion.

Anything that is copy/paste, and especially when the source it lifted from is changing, troubles me -- my preference to actually defer to whatever REST framework is doing, with a minimal override, not copy/paste it.

Comment (madprime) [reply data_import/serializers.py]: Obsoleting this; the function is rewritten from scratch now :+1:

**Pull request merged**